fronts-client/src/services/pandaFetch.ts

```diff
@@ -16,26 +20,55 @@ const pandaFetch = (
   count: number = 0
 ): Promise<Response> =>
   new Promise(
-    async (resolve: (r: Response) => any, reject: (r: Response) => any) => {
-      const res = await fetch(url, {
-        ...options,
-        credentials: 'same-origin',
-      });
+    async (resolve: (r: Response) => any, reject: (e: any) => any) => {
+      try {
+        const res = await fetch(url, {
+          ...options,
+          credentials: 'same-origin',
+        });
 
-      if ((res.status === 419 || res.status === 401) && count < 1) {
-        try {
-          await reEstablishSession(reauthUrl, 5000);
-          const res2 = await pandaFetch(url, options, count + 1);
-          return resolve(res2);
-        } catch (e) {
-          notifications.notify({ message: reauthErrorMessage, level: 'error' });
-          return reject(e);
+        if ((res.status === 419 || res.status === 401) && count < 1) {
+          try {
+            await reEstablishSession(reauthUrl, 5000);
+            const res2 = await pandaFetch(url, options, count + 1);
+            return resolve(res2);
+          } catch (e) {
+            notifications.notify({
+              message: reauthErrorMessage,
+              level: 'error',
+            });
+            Raven.captureException(e, {
+              extra: `Auth issue banner presented to user`,
+            });
+            return reject(
+              isError(e) ? `Auth Issue (${e ? e.toString() : ''})` : e
+            );
+          }
+        } else if (res.status < 200 || res.status >= 300) {
+          // notifications.notify({
+          //   message:
+          //     'Request failed. Your changes may not be saved. Please wait or reload the page.',
+          //   level: 'error',
+          // });
+          Raven.captureException(
+            `'Request failed' banner OUGHT TO BE presented to user, because... ${res.status} ${res.status}`,
+            { extra: res }
+          );
+          return reject(res);
         }
-      } else if (res.status < 200 || res.status >= 300) {
-        return reject(res);
-      }
 
-      return resolve(res);
+        return resolve(res);
+      } catch (error) {
+        // notifications.notify({
+        //   message:
+        //     'Connection issue occurred. Your changes may not be saved. Please wait or reload the page.',
+        //   level: 'error',
+        // });
+        Raven.captureException(error, {
+          extra: `'Connection issue' banner OUGHT TO BE presented to user`,
```

Comment:
Adding text in `extra` seems to result in odd output. From the sentry request:

```
{project: "35467", logger: "javascript", platform: "javascript",…}
breadcrumbs: {,…}
event_id: "df43f97481fa476490c99363b51b983e"
exception: {values: [{type: "TypeError", value: "Failed to fetch",…}],…}
extra: {0: "'", 1: "C", 2: "o", 3: "n", 4: "n", 5: "e", 6: "c", 7: "t", 8: "i", 9: "o", 10: "n", 11: " ",…}
0: "'"
1: "C"
2: "o"
3: "n"
4: "n"
5: "e"
6: "c"
7: "t"
8: "i"
9: "o"
10: "n"
... etc
```

Nesting in the `extra` field seems to solve, suspect they're iterating on 'extra' and spreading into the request:

```
Raven.captureException(error, {
  extra: { message: `'Connection issue' banner OUGHT TO BE presented to user` },
});
```

Reply:
great spot, thanks, I've fixed